### PR TITLE
Custom nonce fixes

### PR DESCRIPTION
**Pull request by danjm**
This PR fixes two issues with the custom nonce feature:
(1) If there is a local failure when submitting a tx, the custom nonce will now be reset, given that the user will be taken away from the confirmation screen
~(2) If there is a pending transaction with a nonce that is higher than the `highestSuggested` tx from the `nonce-tracker` and the user wishes to submit a tx with that nonce (which would be the default one in the nonce field on the confirm screen), attempts to do so would create another tx with the `highestSuggested` nonce.~
(2) The suggested nonce now accounts for locally pending nonces

### CODE REVIEW

File: ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 1]
```diff
@@ -282,7 +285,7 @@ export default class ConfirmTransactionBase extends Component {
                 <TextField
                   type="number"
                   min="0"
-                  placeholder={ this.context.t('nonceFieldPlaceholder') }
```
Comment (Gudahtt) [inline ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js]: The `nonceFieldPlaceholder` message should be removed as well

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -1619,6 +1620,19 @@ module.exports = class MetamaskController extends EventEmitter {
     return pendingNonce
   }
 
+  /**
+   * Returns the next nonce according to the nonce-tracker
+   * @param address {string} - The hex string address for the transaction
+   * @returns Promise<number>
+   */
+  async getNextNonce (address) {
+    const { nonceDetails, releaseLock} = await this.txController.nonceTracker.getNonceLock(address)
```
Comment (Gudahtt) [inline app/scripts/metamask-controller.js]: Isn't `nextNonce` alongside `nonceDetails`, not inside of it?

e.g. 
```suggestion
    const { nextNonce, releaseLock} = await this.txController.nonceTracker.getNonceLock(address)
```

**Review by Gudahtt — CHANGES_REQUESTED** [review]

File: app/scripts/metamask-controller.js (hunk 2 above)
Comment (whymarrh) [reply app/scripts/metamask-controller.js]: Also, nitpick:

```suggestion
    const { nonceDetails, releaseLock } = await this.txController.nonceTracker.getNonceLock(address)
```

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 3]
```diff
@@ -282,7 +282,7 @@ export default class ConfirmTransactionBase extends Component {
...
+                  placeholder={ customNonceValue || nextNonce }
```
Comment (Gudahtt) [inline ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js]: Why is `customNonceValue` used as a placeholder? Presumably if `customNonceValue` is truthy, it'll be set as teh actual value of this field and the placeholder won't be shown at all.

Additionally, I think the `nextNonce` should be removed as a fallback `value` on line 296 if we want to fallback to the tx controller choosing the nonce as usual.

Comment (danjm) [reply ui/app/pages/confirm-transaction-base/confirm-transaction-base.component.js]: fixed in f212ab8e9

**Review by whymarrh — APPROVED** [review]
LGTM

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**